pkg/controller/datavolume-import-controller.go

```diff
@@ -0,0 +1,442 @@
+	}
+	if syncRes.res != nil {
+		return *syncRes.res, nil
+	}
+	//FIXME: pass syncRes instead of args
+	if err := r.reconcileDataVolumePVC(log, syncRes.dv, &syncRes.pvc, syncRes.pvcSpec); err != nil {
+		return reconcile.Result{}, err
+	}
+	return r.reconcileDataVolumeStatus(syncRes.dv, syncRes.pvc, NoClone, r.updateStatusPhase)
+
+}
+
+func (r *DataVolumeImportReconciler) reconcileDataVolumePVC(log logr.Logger, dv *cdiv1.DataVolume, pvc **corev1.PersistentVolumeClaim, pvcSpec *v1.PersistentVolumeClaimSpec) error {
+	if _, dvPrePopulated := dv.Annotations[AnnPrePopulated]; dvPrePopulated {
+		return nil
+	}
+	if *pvc == nil {
+		newPvc, err := r.createPvcForDatavolume(dv, pvcSpec, r.updateCheckpoint)
+		if err != nil {
+			if errQuotaExceeded(err) {
+				r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Pending, dv, nil, NoClone,
+					DataVolumeEvent{
+						eventType: corev1.EventTypeWarning,
+						reason:    ErrExceededQuota,
+						message:   err.Error(),
+					})
+			}
+			return err
+		}
+		*pvc = newPvc
+		return nil
+	}
+	if getSource(*pvc) == SourceVDDK {
+		changed, err := r.getVddkAnnotations(dv, *pvc)
+		if err != nil {
+			return err
+		}
+		if changed {
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: dv.Name, Namespace: dv.Namespace}, dv)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := r.maybeSetMultiStageAnnotation(*pvc, dv); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r DataVolumeImportReconciler) updateStatusPhase(pvc *corev1.PersistentVolumeClaim, dataVolumeCopy *cdiv1.DataVolume, event *DataVolumeEvent) error {
+	phase, ok := pvc.Annotations[AnnPodPhase]
+	if phase != string(corev1.PodSucceeded) {
+		_, ok := pvc.Annotations[AnnImportPod]
+		if !ok || pvc.Status.Phase != corev1.ClaimBound || pvcIsPopulated(pvc, dataVolumeCopy) {
+			return nil
+		}
+		dataVolumeCopy.Status.Phase = cdiv1.ImportScheduled
+	}
+	if !ok {
+		return nil
+	}
+
+	switch phase {
+	case string(corev1.PodPending):
+		// TODO: Use a more generic Scheduled, like maybe TransferScheduled.
+		dataVolumeCopy.Status.Phase = cdiv1.ImportScheduled
+		event.eventType = corev1.EventTypeNormal
+		event.reason = ImportScheduled
+		event.message = fmt.Sprintf(MessageImportScheduled, pvc.Name)
+	case string(corev1.PodRunning):
+		// TODO: Use a more generic In Progess, like maybe TransferInProgress.
+		dataVolumeCopy.Status.Phase = cdiv1.ImportInProgress
+		event.eventType = corev1.EventTypeNormal
+		event.reason = ImportInProgress
+		event.message = fmt.Sprintf(MessageImportInProgress, pvc.Name)
+	case string(corev1.PodFailed):
+		event.eventType = corev1.EventTypeWarning
+		event.reason = ImportFailed
+		event.message = fmt.Sprintf(MessageImportFailed, pvc.Name)
+	case string(corev1.PodSucceeded):
+		if _, ok := pvc.Annotations[AnnCurrentCheckpoint]; ok {
+			if err := r.updatesMultistageImportSucceeded(pvc, dataVolumeCopy); err != nil {
+				return err
+			}
+			//FIXME: why needed? who sets it to succeeded in the end?
```

Comment:
Because the importer pod succeeds for each delta, so the annotation is set to succeeded while the entire import is not 100% complete, this is why we then set it to paused instead to indicate more deltas are coming.

All this handling is pretty specific to warm imports, which is why I suggested maybe make it its own controller to separate the code.